File: rust/src/models/create_product_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{product_family_id::ProductFamilyId, product_fee_structure::ProductFeeStructure};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CreateProductRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub catalog: Option<bool>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    pub fee_structure: ProductFeeStructure,

    pub name: String,

    pub product_family_id: ProductFamilyId,
}

impl CreateProductRequest {
    pub fn new(
        fee_structure: ProductFeeStructure,
        name: String,
        product_family_id: ProductFamilyId,
    ) -> Self {
        Self {
            catalog: None,
            description: None,
            fee_structure,
            name,
            product_family_id,
        }
    }
}

File: typescript/src/models/createProductRequest.ts
// this file is @generated
import { type ProductFamilyId, ProductFamilyIdSerializer } from "./productFamilyId";
import {
  type ProductFeeStructure,
  ProductFeeStructureSerializer,
} from "./productFeeStructure";

export interface CreateProductRequest {
  catalog?: boolean;

  description?: string | null;

  feeStructure: ProductFeeStructure;

  name: string;

  productFamilyId: ProductFamilyId;
}

export const CreateProductRequestSerializer = {
  _fromJsonObject(object: any): CreateProductRequest {
    return {
      catalog: object["catalog"],
      description: object["description"],
      feeStructure: ProductFeeStructureSerializer._fromJsonObject(
        object["fee_structure"]
      ),
      name: object["name"],
      productFamilyId: ProductFamilyIdSerializer._fromJsonObject(
        object["product_family_id"]
      ),
    };
  },

  _toJsonObject(self: CreateProductRequest): any {
    return {
      catalog: self.catalog,
      description: self.description,
      fee_structure: ProductFeeStructureSerializer._toJsonObject(self.feeStructure),
      name: self.name,
      product_family_id: ProductFamilyIdSerializer._toJsonObject(self.productFamilyId),
    };
  },
};
